spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/TupleJsonMessageConverter.java

```diff
@@ -79,4 +74,16 @@ public Object convertFromInternal(Message<?> message, Class<?> targetClass, Obje
 		return json;
 	}
 
+	@Override
+	public Object convertFromInternal(Message<?> message, Class<?> targetClass, Object conversionHint) {
+		String source;
+		if (message.getPayload() instanceof byte[]) {
+			source = new String((byte[]) message.getPayload());
```

Comment:
`UTF-8` is the default. We can extract the encoding from the content type I suppose.